rust/protocol/src/sealed_sender.rs

```diff
@@ -735,7 +742,11 @@ pub async fn sealed_sender_encrypt_from_usmc<R: Rng + CryptoRng>(
 
     let message_data = crypto::aes256_ctr_hmacsha256_encrypt(
         usmc.serialized()?,
-        &static_keys.cipher_key()?,
+        &static_keys.cipher_key().and_then(|key| {
+            key.try_into().map_err(|_: TryFromSliceError| {
+                SignalProtocolError::BadKeyLength(curve::KeyType::Curve25519, key.len())
+            })
+        })?,
```

Comment:
This seems like a type system failure; we should never get another length of key from the intermediate step here of calculating static keys.

Reply:
Yes! I have refactored these methods to return static-sized arrays in all possible cases, which allows us to avoid re-checking these lengths as often.